pkg/armhelpers/loganalytics.go

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+package armhelpers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	oi "github.com/Azure/azure-sdk-for-go/services/preview/operationalinsights/mgmt/2015-11-01-preview/operationalinsights"
+	om "github.com/Azure/azure-sdk-for-go/services/preview/operationsmanagement/mgmt/2015-11-01-preview/operationsmanagement"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+// EnsureDefaultLogAnalyticsWorkspace ensures the default log analytics exists corresponding to specified location in selected subscription
+func (az *AzureClient) EnsureDefaultLogAnalyticsWorkspace(ctx context.Context, resourceGroup, location string) (workspaceResourceID string, err error) {
```

Comment:
same comment here about unit testing this new file

Reply:
> same comment here about unit testing this new file

added the unit tests to cover these methods.